lib/terminal.js

```diff
@@ -38,7 +38,13 @@ export default class Terminal {
     })
 
     this.channel.onClose(e => this.emit('close', e))
-    this.channel.onError(e => this.emit('error', e))
+
+    this.channel.onError((e) => {
+      this.channel.leave()
+      this.socket.reconnectTimer.callback = () => { } // noops to prevent auto reconnects
```

Comment:
This is pretty gnarly. There's no public API from the looks of it ( @ktravers and I scoured the JS source code to look) for disabling the exponential backoff reconnect feature from the Phoenix socket client. All `Socket` objects created are initialized with a `Timer` object that on disconnects will make a callback to reset the socket by calling `disconnect` and then `connect`. That code is here: https://github.com/phoenixframework/phoenix/blob/71b1c2e85960d4124cec86bc287956ed6c69239a/priv/static/phoenix.js#L759-L763

We could provide a `reconnectAfterMs` callback that returns an extremely large number, but that seems kind of janky, although it might be less gross than reaching into the socket's `reconnectTimer` and reassigning the callback to a noop.

I'll open an issue on the phoenix repo to see whether an option can be provided to disable autoreconnect, or if there is a way and I'm just not figuring it out correctly